### PR TITLE
Many updates

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] Implement finetuning with aggregates for the EuroSat and TreeSat tasks
- [x] Update the Presto model to accept varying masked levels within a batch
- [x] Additionally evaluate against the Sen2Agri crop type classification task  

Paper with the updated results to be added to arxiv. I'll update the link here once that's done.

Changes were co-authored with @rubencart.

### CODE REVIEW

File: mosaiks.py [hunk 1]
```diff
@@ -101,11 +94,11 @@ def load_dataset(url, shuffle_on_load):
 
     logger.info("Loading evaluation tasks")
     eval_task_list: List[EvalTask] = [
-        *[FuelMoistureEval(seeds=[s]) for s in eval_seeds],
-        *[AlgaeBloomsEval(seeds=[s]) for s in eval_seeds],
-        *[CropHarvestEval("Kenya", seeds=[s]) for s in eval_seeds],
-        *[CropHarvestEval("Togo", seeds=[s]) for s in eval_seeds],
-        *[CropHarvestEval("Brazil", seeds=[s]) for s in eval_seeds],
+        FuelMoistureEval(seed=seed),
```
Comment (rubencart) [inline mosaiks.py:97]: So for mosaiks you start a wandb run per seed? ([line 91](https://github.com/nasaharvest/presto/pull/32/files#diff-8943f236dd39782f2e4786a6986d26817c8f66c9d4c11804f17d9f6aeefaf684R91))

File: presto/eval/eurosat_eval.py [hunk 2]
```diff
@@ -152,12 +157,11 @@ def __init__(
         assert IMAGE_SIZE % input_patch_size == 0
         self.num_patches_per_dim = int(IMAGE_SIZE / input_patch_size)
         self.num_patches = self.num_patches_per_dim**2
+        self.batch_size = min(self.num_patches * 64, 40920)
```
Comment (rubencart) [inline presto/eval/eurosat_eval.py:160]: Still confusing to me why you have `64` here and the same line but with `self.num_patches * 36` in TreeSatEval

File: presto/eval/treesat_eval.py [hunk 3]
```diff
@@ -347,51 +547,38 @@ def finetuning_results(
...
+    def finetune_with_aggregate(
+        self, pretrained_model, mask: Optional[np.ndarray], aggregate: str
+    ) -> FineTuningModel:
+        hparams = Hyperparams(batch_size=self.batch_size)
+        model = self._construct_finetuning_model_with_aggregates(pretrained_model, aggregate)
+        opt = AdamW(model.parameters(), lr=hparams.lr, weight_decay=hparams.weight_decay)
+
+        loss_fn = nn.CrossEntropyLoss(reduction="mean")
+        ds = TreeSatDataset(self.input_patch_size, split="train")
+        train_ds, val_ds = ds.train_val_split(val_ratio=0.1, seed=self.seed)
```
Comment (rubencart) [inline presto/eval/treesat_eval.py:566]: This means you'll end up with a different validation set per seed?

**Review by rubencart — APPROVED** [review]
Looks good!

File: presto/eval/treesat_eval.py (hunk 3 above)
Comment (gabrieltseng) [reply presto/eval/treesat_eval.py:566]: Yes, but not different test sets.

File: mosaiks.py (hunk 1 above)
Comment (gabrieltseng) [reply mosaiks.py:97]: yes, because each seed yields a different MOSAIKS model.

File: presto/eval/treesat_eval.py (hunk 3 above)
Comment (rubencart) [reply presto/eval/treesat_eval.py:566]: I get that this is another way to use seeds to 'cancel' random effects during training, but from the perspective of making experiments comparable, it seems more practical to use the same validation set? + other tasks use identical val sets for different seeds too?
